src/features/broker-protection/actions/extract.js

```diff
@@ -41,18 +43,21 @@ import { ProfileUrlExtractor } from '../extractors/profile-url.js'
  * @param {Action} action
  * @param {Record<string, any>} userData
  * @param {Document | HTMLElement} root
- * @return {import('../types.js').ActionResponse}
+ * @return {Promise<import('../types.js').ActionResponse>}
  */
-export function extract (action, userData, root = document) {
+export async function extract (action, userData, root = document) {
     const extractResult = extractProfiles(action, userData, root)
 
     if ('error' in extractResult) {
         return new ErrorResponse({ actionID: action.id, message: extractResult.error })
     }
 
-    const filtered = extractResult.results
+    const filteredPromises = extractResult.results
         .filter(x => x.result === true)
         .map(x => aggregateFields(x.scrapedData))
+        .map(profile => generateProfileId(profile, action?.profile?.profileUrl))
```

Comment:
I think we can make this pattern more general - I can see the use-case, it's all about a transform that happens, but it can only be computed *after* you've aggregated all the fields etc. So, I think we can have a list of  `AsyncProfileTransform`s to apply in a loop. that way, we can co-locate more profile url/hash stuff. 

See: https://github.com/duckduckgo/content-scope-scripts/commit/0c9bdfcd0c7b48915a1fc864c21e88555bdedf9d